components/function-controller/internal/controllers/serverless/function_controller.go

```diff
@@ -94,9 +95,10 @@ func (r *FunctionReconciler) updateStatus(
 
```

Comment:
I think we should also check if `instance.Status.Commit == commit` in line 92, otherwise if someone uses branch name as reference, the Repositories will be equal and the status won't be updated with the new `.Status.Commit`.